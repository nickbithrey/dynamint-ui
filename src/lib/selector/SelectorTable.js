import React from 'react';
import PropTypes from 'prop-types';
import Table from '../Table';
import { MarqueeSelection } from 'office-ui-fabric-react/lib/MarqueeSelection';
import { Selection } from 'office-ui-fabric-react/lib/DetailsList';

const SelectorTable = ({items, load, columns, selection}) => (
	<MarqueeSelection selection={selection}>
		<Table list={items} load={load} columns={columns} selection={selection}></Table>
	</MarqueeSelection>
);

SelectorTable.propTypes = {
	itms: PropTypes.array,
	load: PropTypes.func.isRequired,
	columns: PropTypes.arrayOf(PropTypes.object).isRequired,
	selection: PropTypes.object.isRequired
};

export default SelectorTable;