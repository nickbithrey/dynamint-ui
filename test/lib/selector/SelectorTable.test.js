import React from 'react';
import { shallow } from 'enzyme';

import SelectorTable from '~/lib/selector/SelectorTable';

describe('<SelectorTable /> rendering', () => {

	const props = {
		items: [
			{f: 'v', f2: 'v2'},
			{f: 'v3', f2: 'v4'}
		],
		load: jest.fn(() => []),
		columns: [
			{field: 'f'},
			{field: 'f2'}
		],
		selection: {
			count: 0
		}
	};
	
	let wrapper;
	
	beforeEach(() => {
		wrapper = shallow( <SelectorTable {...props} /> );
	});

	it('can render standard component with records', () => {
		expect(wrapper).toMatchSnapshot();
	});
		
});