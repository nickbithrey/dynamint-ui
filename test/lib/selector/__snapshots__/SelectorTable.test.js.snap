// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectorTable /> rendering can render standard component with records 1`] = `
<StyledMarqueeSelectionBase
  selection={
    Object {
      "count": 0,
    }
  }
>
  <Table
    columns={
      Array [
        Object {
          "field": "f",
        },
        Object {
          "field": "f2",
        },
      ]
    }
    list={
      Array [
        Object {
          "f": "v",
          "f2": "v2",
        },
        Object {
          "f": "v3",
          "f2": "v4",
        },
      ]
    }
    load={[MockFunction]}
    selection={
      Object {
        "count": 0,
      }
    }
  />
</StyledMarqueeSelectionBase>
`;
